pytensor/xtensor/shape.py

```diff
@@ -301,3 +301,125 @@ def make_node(self, *inputs: Variable) -> Apply:
+            return Apply(self, [x], [x])
+
+        # Check if dimension already exists
+        if self.dim in x.type.dims:
+            raise ValueError(f"Dimension {self.dim} already exists")
+
+        # Add new dimension at the beginning
+        new_dims = [self.dim, *list(x.type.dims)]
+        new_shape = [self.size, *list(x.type.shape)]
+
+        output = xtensor(
+            dtype=x.type.dtype,
+            dims=tuple(new_dims),
+            shape=tuple(new_shape),
+        )
+        return Apply(self, [x], [output])
+
+
+def expand_dims(x, dim: str):
+    """Add a new dimension to an XTensorVariable.
+
+    Parameters
+    ----------
+    x : XTensorVariable
+        The input tensor
+    dim : str
+        The name of the new dimension
+
+    Returns
+    -------
+    XTensorVariable
+        A new tensor with the expanded dimension
+    """
+    return ExpandDims(dim=dim)(x)
+
+
+class Squeeze(XOp):
+    """Remove dimensions of size 1 from an XTensorVariable.
+
+    Parameters
+    ----------
+    dim : str or None or iterable of str
+        The name(s) of the dimension(s) to remove. If None, all dimensions of size 1 will be removed.
+    """
+
+    def __init__(self, dim=None):
+        self.dim = dim
+
+    def make_node(self, x):
+        x = as_xtensor(x)
+
+        # Convert single dimension to iterable for consistent handling
+        dims_to_remove = [self.dim] if isinstance(self.dim, str) else self.dim
+
+        if dims_to_remove is not None:
+            # Validate dimensions exist and have size 1
+            for dim in dims_to_remove:
+                if dim not in x.type.dims:
+                    raise ValueError(f"Dimension {dim} not found")
+                dim_idx = x.type.dims.index(dim)
+                # Only raise an error if the shape is statically known and not 1.
+                # If the shape is None (symbolic), defer the error to runtime.
+                if x.type.shape[dim_idx] is not None and x.type.shape[dim_idx] != 1:
+                    raise ValueError(
+                        f"Dimension {dim} has size {x.type.shape[dim_idx]}, not 1"
+                    )
+            # Get indices of dimensions to remove
+            dim_indices = [x.type.dims.index(dim) for dim in dims_to_remove]
+        else:
+            # Find all dimensions of size 1
+            dim_indices = [i for i, s in enumerate(x.type.shape) if s == 1]
+            if not dim_indices:
+                raise ValueError("No dimensions of size 1 to remove")
+
+        # Create new dimensions and shape lists
+        new_dims = [d for i, d in enumerate(x.type.dims) if i not in dim_indices]
+        new_shape = [s for i, s in enumerate(x.type.shape) if i not in dim_indices]
+
+        output = xtensor(
+            dtype=x.type.dtype, shape=tuple(new_shape), dims=tuple(new_dims)
+        )
+        return Apply(self, [x], [output])
+
+
+def squeeze(x, dim=None):
+    """Remove dimensions of size 1 from an XTensorVariable.
```

Comment:
Add a note that this deviates from numpy/xarray. Similar to what we have here: https://github.com/pymc-devs/pytensor/blob/4c8c8b6e35f503ebde99a94ea10c3d95af9f98af/pytensor/tensor/extra_ops.py#L604-L606

The first point is actually not true anymore, so don't copy it